src/main/java/com/google/edith/servlets/MailServlet.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edith.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.Properties;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/notifications")
+public class MailServlet extends HttpServlet {
+
+  private final boolean mockSend;
+
+  MailServlet() {
+    this.mockSend = false;
+  }
+
+  public MailServlet(boolean mockSend) {
+    this.mockSend = mockSend;
+  }
```

Comment:
Instead of referencing mocks in the constructor as a boolean variable, an interface should be used: 

```java

private Mailer mailer;

interface Mailer {
  void sendMail(String subject, String body);
}

public MailServlet() {
  this.mailer = this::sendSimpleMail;
}

private MailServlet(Mailer mailer) {
  this.mailer = mailer;
}

static MailServlet withMockMailer() {
  return new MailServlet(MailServlet::mockSendMail);
}

private static void sendSimpleMail(String subject, String body) {
  ...
}

private static void mockSendMail(String subject, String body) {
  ...
}
```

then instead of checking `if (mockSend)`, the Mailer can be used directly